x-pack/plugins/painless_lab/public/application/context.tsx

```diff
@@ -8,50 +8,89 @@ import React, { createContext, ReactNode, useState, useContext } from 'react';
 import { HttpSetup } from 'src/core/public';
 
 import { Links } from '../links';
+import { exampleScript, painlessContextOptions } from './constants';
+import { Store, Payload, Validation } from './types';
 
-import { initialState, Store } from './store';
-import { PAINLESS_LAB_KEY } from './constants';
+interface AppContextProviderArgs {
+  children: ReactNode;
+  value: {
+    http: HttpSetup;
+    links: Links;
+  };
+}
 
 interface ContextValue {
-  state: Store;
-  updateState: (nextState: (s: Store) => Partial<Store>) => void;
+  store: Store;
+  updatePayload: (changes: Partial<Payload>) => void;
   services: {
     http: HttpSetup;
   };
   links: Links;
 }
 
+const initialPayload = {
```

Comment:
@jloleysens What do you think of bringing this into this file? It's not consumed elsewhere so I think it creates noise to extract it. If we want to break this file's code up into multiple files, I'd prefer to break it out into its own module (a directory with an index file) to make it clear what's intended for consumption elsewhere.

Reply:
Sure, I am happy with this change. I had the `store.ts` file separate so that if we choose to use a reducer in future we can just add the reducer to the store file.

Reply:
I do think if we start to expand more on validators it would be nice to have a separate file for these. Ideally I like to keep the `context` file as minimal as possible so that it only has the concern setting up the context. Where, at the moment, it houses store setup and validation logic - perhaps worth creating the folder structure you were speaking about. That way future engineers know where to put things?

Reply:
OK, I'll create a folder now. In terms of validation, I would expect us to try using the form hook lib if we want to add more rules in the future.